src/tokens/tokens.service.ts

```diff
@@ -199,7 +209,9 @@ export class TokensService {
     public http: HttpService,
     private eventstream: EventStreamService,
     private proxy: EventStreamProxyGateway,
-  ) {}
+  ) {
+    this.uriCache = new LRUCache({ max: 500 });
```

Comment:
nit: this isn't really a cache of URIs, but a cache of whether a given contract has URI support - right?